module/VuFind/src/VuFind/Db/Service/UserService.php

```diff
@@ -63,10 +63,31 @@ public function __construct(protected User $userTable)
      *
      * @param string $id ID.
      *
-     * @return UserEntityInterface
+     * @return ?UserEntityInterface
      */
     public function getUserById($id)
     {
         return $this->userTable->getById($id);
     }
+
+    /**
+     * Retrieve a user object from the database based on the given field.
+     *
+     * @param string $fieldName  Field name
+     * @param string $fieldValue Field value
+     *
+     * @return ?UserEntityInterface
+     */
+    public function getUserByField($fieldName, $fieldValue)
+    {
+        switch ($fieldName) {
+            case 'id':
+                return $this->userTable->getById($fieldValue);
+            case 'username':
+                return $this->userTable->getByUsername($fieldValue, false);
+            case 'cat_id':
+                return $this->userTable->getByCatalogId($fieldValue);
+        }
+        return null;
```

Comment:
Might it be better to throw an exception complaining about an unsupported field if we reach this point in the method?

Reply:
Yes, that's a good idea. Done